toolkit/Yara.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef DARWIN_YARA_HPP
+#define DARWIN_YARA_HPP
+
+extern "C" {
+#include <yara.h>
+}
+
+#include <memory>
+#include <vector>
+#include <set>
+#include "rapidjson/document.h"
+
+namespace darwin {
+
+    namespace toolkit {
+
+        class YaraEngine {
+        public:
+            YaraEngine(bool fastmode = true, int timeout = 0);
+            ~YaraEngine();
+
+            bool Init(YR_RULES *rules);
+            int ScanData(std::vector<unsigned char> &data, unsigned int &certitude);
+            rapidjson::Document GetResults();
+
+        private:
+            static int ScanOrImportCallback(int message, void *messageData, void *userData);
+            void AddRuleToMatch(YR_RULE *rule);
```

Comment:
It would be nice to include the Doxygen formated documentation.

Reply:
Done in ef19641